rancid-tomatillos/src/App/App.test.js

```diff
@@ -1,9 +1,39 @@
-import React from 'react';
-import { render } from '@testing-library/react';
-import App from './App';
+import React from "react";
+import ReactDOM from "react-dom";
+import App from "./App";
+import {render, screen} from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import {Router} from "react-router-dom";
+import {createMemoryHistory} from "history";
+import "@testing-library/jest-dom/extend-expect";
+import "@testing-library/jest-dom";
 
-test('renders learn react link', () => {
-  const { getByText } = render(<App />);
-  const linkElement = getByText(/learn react/i);
-  expect(linkElement).toBeInTheDocument();
+describe.only("App", () => {
+    test("User is redirected to login page on link click", () => {
+        const history = createMemoryHistory();
+        render(
+            <Router history={history}>
+                <App />
+            </Router>
+        );
+        userEvent.click(screen.getByRole("link", {name: "Login"}));
+        expect(screen.getByText(/login/i)).toBeInTheDocument()
+        expect(screen.getByRole("button", {name: "Enter User Details"})).toBeInTheDocument()
+    });
+    test.only("User can log in", () => {
+        const history = createMemoryHistory();
+        render(
+            <Router history={history}>
+                <App />
+            </Router>
+        );
+        userEvent.click(screen.getByRole("link", {name: "Login"}));
+        userEvent.type(screen.getByPlaceholderText("email"), "marge@turing.io");
+        userEvent.type(screen.getByPlaceholderText("password"), "test");
+        expect(screen.getByPlaceholderText('email')).toHaveValue('marge@turing.io')
+        expect(screen.getByPlaceholderText('password')).toHaveValue('test')
+        screen.debug()
+        userEvent.click(screen.getByRole("button", {name: "Enter User Details"}));
+        //expect(screen.getByText("incorrect login")).toBeInTheDocument()
```

Comment:
Things to figure out:
1) Break down tests into smaller individual tests -- what are we testing for with each step of logging in>
2) How to get the test to render/reroute us to the right page after a user enters correct login details vs incorrect login details